synapse/storage/databases/main/event_federation.py

```diff
@@ -1530,6 +1530,78 @@ def _record_event_failed_pull_attempt_upsert_txn(
 
         txn.execute(sql, (room_id, event_id, 1, self._clock.time_msec(), cause))
 
+    @trace
+    async def filter_event_ids_with_pull_attempt_backoff(
```

Comment:
The name here is ambiguous at the moment. It basically does the opposite of what I expect. For `filter_event_ids_with_pull_attempt_backoff`, I would expect it to filter out events that we've failed to pull before recently leaving behind only the event_ids which are ok to pull again. But instead, it filters down to the event_ids to ignore and not pull again right now.

What's a good name for this?

We could make it return the opposite, events which are ok to pull again but that isn't what our usage needs exactly.

And we would either have to do some extra `set`/`list` comprehension to return the inverse or adjust the SQL to create a temporary `VALUES` table to query against.

I've opted to leave it simple in case someone has a good name for the current behavior?

Reply:
Re-commented on the diff since it disappeared even though not oudated, https://github.com/matrix-org/synapse/pull/13816#discussion_r987741403